### PR TITLE
Fix badge size issue when a number larger than 3 digits is used

### LINKED ISSUE #34015: Design issue coming to order admin home page
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When you open the order admin home page, then the circle design is created in the counts like order counts and stock counts, that design counts after 3 digits, and the design of the circle is deteriorating.

CleanShot 2022-07-21 at 15 43 29@2x

### Expected behavior

CleanShot 2022-07-21 at 15 53 30@2x

### Actual behavior

CleanShot 2022-07-21 at 15 43 29@2x

### Steps to reproduce

1. Activate the WooCommerce plugin by installing it on your site
2. Go to order admin home page
3. Inspect and write down more than 3 digits of order counts

### WordPress Environment

Woocommerce Version: 6.7.0
WordPress version: 6.0.1

PC
Mac OS: Version 12.4
Google Chrome: Version 103.0.5060.114 (Official Build) (arm64)

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #34015 .

This PR fixes badge size issue when a number larger than 3 digits is used.

Credit to @aezazs-multidots for the fix.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Go to `WooCommerce -> Home`
3. Right click on the badge next to `Thigns to do next` and choose `Inspect`
4. Change the number to 9999
5. Confirm the badge expands.

Screen Shot 2023-10-05 at 3 24 47 PM

Before:

Screen Shot 2023-10-05 at 3 25 06 PM

After:

Screen Shot 2023-10-05 at 3 25 30 PM

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Fix badge size issue when a number larger than 3 digits is used.

#### Comment 

</details>

### CODE REVIEW

File: packages/js/components/src/badge/style.scss [hunk 1]
```diff
@@ -8,6 +8,7 @@
 	font-size: 14px;
 	line-height: 27px;
 	align-items: center;
-	width: 32px;
-	height: 28px;
+	width: auto;
+	height: auto;
```
Comment (chihsuan) [inline packages/js/components/src/badge/style.scss]:
```suggestion
    min-width: 32px;
    height: 28px;
```

It seems to cause the normal styling to be a bit different. 

Screenshot 2023-10-06 at 16 10 18

Fixed:
Screenshot 2023-10-06 at 16 10 22

**Review by chihsuan — DISMISSED** [review]
Thanks for the fix! Just left a suggestion.

**Review by rjchow — APPROVED** [review]
Chi's suggestion looks good!

**Review by psealock — APPROVED** [review]
Works well, thanks @moon0326 🚀

**Pull request merged**